uber/site_sections/registration.py

```diff
@@ -643,7 +643,11 @@ def new_checkin(self, session, id, badge_num='', ec_phone='', message='', group=
         elif existing:
             message = '{a.badge} already belongs to {a.full_name}'.format(a=existing[0])
         else:
-            badge_type, message = get_badge_type(badge_num)
+            message = check_range(badge_num, attendee.badge_type)
+            if not message:
+                maybe_dupe = session.query(Attendee).filter_by(badge_num=badge_num, badge_type=attendee.badge_type)
+                if maybe_dupe.count():
+                    message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
             if not message:
                 attendee.badge_type, attendee.badge_num = badge_type, badge_num
```

Comment:
All I can tell you is that it doesn't error on my ONE test case locally.
Still no need for the badge_type, so I removed it.

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

On Thu, Feb 11, 2016 at 5:05 PM, Eli Courtwright <notifications@github.com>
wrote:

> In uber/site_sections/registration.py
> <https://github.com/magfest/ubersystem/pull/1690#discussion_r52675137>:
>
> > @@ -643,7 +643,11 @@ def new_checkin(self, session, id, badge_num='', ec_phone='', message='', group=
> >          elif existing:
> >              message = '{a.badge} already belongs to {a.full_name}'.format(a=existing[0])
> >          else:
> > -            badge_type, message = get_badge_type(badge_num)
> > +            message = check_range(badge_num, attendee.badge_type)
> > +            if not message:
> > +                maybe_dupe = session.query(Attendee).filter_by(badge_num=badge_num, badge_type=attendee.badge_type)
> > +                if maybe_dupe.count():
> > +                    message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
> >              if not message:
> >                  attendee.badge_type, attendee.badge_num = badge_type, badge_num
>
> I think it must error in that case - I just read through the whole
> function in your branch at
>
> https://github.com/magfest/ubersystem/blob/Make_Recent_ATD_Reg_enforce_badge_types/uber/site_sections/registration.py
> and badge_type is never defined there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1690/files#r52675137>.
>